.github/workflows/rust.yml

```diff
@@ -18,8 +18,12 @@ jobs:
       - uses: Swatinem/rust-cache@v2
       - name: Format
         run: cargo fmt --all -- --check
-      - name: Build
+      - name: Build minimal
+        run: cargo build --verbose --no-default-features
+      - name: Build with default features
         run: cargo build --verbose
+      - name: Build with all features
+        run: cargo build --verbose --all-features
```

Comment:
It's somewhat standard that a Rust crate should be able to build with `--all-features`, for instance `docs.rs` assumes this.

However, this will need to be tweaked if [mutually exclusive features](https://doc.rust-lang.org/cargo/reference/features.html#mutually-exclusive-features) are added in future, and [docs.rs metadata](https://docs.rs/about/metadata) will have to be added.